src/dbnode/storage/bootstrap/types.go

```diff
@@ -224,8 +224,13 @@ type Namespace struct {
 // NamespaceRunOptions are the run options for a bootstrap process run.
 type NamespaceRunOptions struct {
 	// ShardTimeRanges are the time ranges for the shards that should be fulfilled
-	// by the bootstrapper.
+	// by the bootstrapper. This changes each bootstrapper pass as time ranges are fulfilled.
 	ShardTimeRanges result.ShardTimeRanges
+	// TargetShardTimeRanges are the original target time ranges for shards and does not change
+	// each bootstrapper pass.
+	// NB(bodu): This is used by the commit log bootstrapper as it needs to run for the entire original
+	// target shard time ranges.
+	TargetShardTimeRanges result.ShardTimeRanges
```

Comment:
This is the additional option I added to capture the original target time ranges for shards that does not change per bootstrapper pass.